File: Chapter03/hanoiTower.py
# 汉诺塔
def move(x:str,y:str):
    print("{0}===>{1}".format(x,y))
def hanoiTower(n:int,a:str,b:str,c:str):
    if n==1:
        move(a,c)
    else:
        hanoiTower(n-1,a,c,b)
        move(a,c)
        hanoiTower(n-1,b,a,c)

hanoiTower(3,"A塔","B塔","C塔")

File: Chapter03/hanoiTower.js
/* 汉诺塔 */
const move=(x,y)=>{
    console.log(`${x}===>${y}`);
}

const hanoiTower=(n,a,b,c)=>{
    if(n==1){
        move(a,c);
    }else{
        hanoiTower(n-1,a,c,b);
        move(a,c);
        hanoiTower(n-1,b,a,c);
    }
}

hanoiTower(3,"A塔","B塔","C塔");